src/app/core/services/data.service.ts

```diff
@@ -62,21 +64,23 @@ export class DataService {
     this.subject.next(this.love ? lovedVideos : this.userVideosList);
   }
 
-  public deleteVideo(id: string): void {
-    this.userVideosList = this.userVideosList.filter((video) => video.videoId !== id);
-    this.subject.next(this.userVideosList);
-    this.localStorageService.saveToLocalStorage(this.userVideosList);
+  public deleteVideo(videoId: string): void {
+    this.lastDeletedVideo = this.userVideosList.find((video) => video.videoId === videoId);
+    this.userVideosList = this.userVideosList.filter((video) => video.videoId !== videoId);
```

Comment:
hmm  to w takim razie możesz wykorzystać findIndex tak aby ograć to w jednym kroku 
const lastDeletedVideoIndex = this.userVideosList.find((video) => video.videoId === videoId);
this.lastDeletedVideo =  this.userVideosList[lastDeletedVideoIndex]
this.userVideosListsplice(lastDeletedVideoIndex, 1)